backend/main/src/User/Domain/User.ts

```diff
@@ -3,26 +3,20 @@ import { DomainObjectError } from "src/Shared";
 
 interface IUserProps {
 	name: string;
-	isAdmin: boolean;
-}
-
-interface IUserPropsInput extends Omit<IUserProps, "isAdmin"> {
-	isAdmin?: boolean;
 }
 
+/**
+ * This class is user class.
+ * The name of user must be less than equal to 30.
+ */
 export class User extends AggregateRoot<string, IUserProps> {
-	// username must be shorter than 100.
-	static create(
-		id: UserId,
-		props: IUserPropsInput,
-	): Result<User, UserDomainError> {
-		if (props.name.length > 100) {
+	static create(id: UserId, props: IUserProps): Result<User, UserDomainError> {
```

Comment:
These changes require the following code to be modified. I'm not sure if it is the best way, but I mean, a type guard seems needed.

> backend/main/src/User/UseCases/UpdateUserUseCase/UpdateUserNameUseCase.ts: 51

```typescript
if (!nameUpdatedUser.ok) {
	return Err(nameUpdatedUser.error);
}

await this.userRepository.update(nameUpdatedUser.value);
```